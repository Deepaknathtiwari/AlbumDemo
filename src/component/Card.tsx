import React from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';
import { vw, fullWidth } from '../service/dimension';
import Image from './Image';
import { parser } from '../service/parser';

interface Props {
  rowData: { item: any; index: number };
  handleNavigation: (item: any) => void;
}

const AlbumCard = React.memo(function Card(props: Props) {
  return (
    <TouchableOpacity
      onPress={() => props.handleNavigation(props.rowData.item)}
      style={styles.container}>
      <Image
        source={{
          uri: parser(props.rowData.item.artworkUrl100, 'source/300x300'),
        }}
        style={styles.imageStyle}
        data-test="Album-card-component"
      />
      <Text
        style={styles.title}
        numberOfLines={
          1
        }>{`${props.rowData.item?.trackName?.toUpperCase()}`}</Text>
    </TouchableOpacity>
  );
});

export default AlbumCard;

const styles = StyleSheet.create({
  container: {
    width: fullWidth / 2 - 10,
    marginHorizontal: 5,
    marginVertical: 15,
    borderRadius: 5,
  },
  imageStyle: {
    borderRadius: 5,
    height: vw(175),
    width: vw(175),
  },
  title: {
    marginTop: 5,
    fontSize: vw(16),
    fontWeight: 'bold',
  },
});
