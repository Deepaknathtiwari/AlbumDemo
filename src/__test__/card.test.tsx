/* eslint-disable @typescript-eslint/no-unused-vars */
import React from 'react';
import { shallow } from 'enzyme';
import Card from '../component/Card';
const findByTestAttr = (component: any, attr: any) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};
describe('Album Card Component', () => {
  const wrapperCardComponent = shallow(
    <Card
      rowData={{ item: {}, index: 0 }}
      handleNavigation={(item: any) => {}}
    />,
  );
  it('Album snapshot', () => {
    expect(wrapperCardComponent).toMatchSnapshot('GP Card snap');
  });
  it('Album card render', () => {
    const component = findByTestAttr(
      wrapperCardComponent,
      'Album-card-component',
    );
    expect(component.length).toBe(1);
  });
});
