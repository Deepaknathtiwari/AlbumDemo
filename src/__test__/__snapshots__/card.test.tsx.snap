// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album Card Component Album card snapshot: Album card snapshot 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Object {
      "borderRadius": 5,
      "marginHorizontal": 5,
      "marginVertical": 15,
      "width": 365,
    }
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "borderRadius": 5,
        "height": 350,
        "width": 350,
      }
    }
  />
  <Text
    numberOfLines={1}
    style={
      Object {
        "fontSize": 32,
        "fontWeight": "bold",
        "marginTop": 5,
      }
    }
  >
    undefined
  </Text>
</ForwardRef>
`;

exports[`Album Card Component Album snapshot: GP Card snap 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Object {
      "borderRadius": 5,
      "marginHorizontal": 5,
      "marginVertical": 15,
      "width": 365,
    }
  }
>
  <Image
    data-test="Album-card-component"
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "borderRadius": 5,
        "height": 350,
        "width": 350,
      }
    }
  />
  <Text
    numberOfLines={1}
    style={
      Object {
        "fontSize": 32,
        "fontWeight": "bold",
        "marginTop": 5,
      }
    }
  >
    undefined
  </Text>
</ForwardRef>
`;
